esphome/components/external_eeprom/external_eeprom.cpp

```diff
@@ -0,0 +1,443 @@
+  double val;
+  this->read(memaddr, (uint8_t *) &val, sizeof(double));
+  return val;
+}
+/// @brief Bulk read from the device
+/// @note breaking up read amt into 32 byte chunks (can be overriden with setI2Cbuffer_size)
+/// @note Handles a read that straddles the 512kbit barrier
+/// @param memaddr is the starting location to read
+/// @param buff is the pointer to an array of bytes that will be used to store the data received
+/// @param buffer_size is the size of the buffer and also the number of bytes to be read
+void ExtEepromComponent::read(uint32_t memaddr, uint8_t *buff, uint16_t buffer_size) {
+  ESP_LOGVV(TAG, "Read %d bytes from address %d", buffer_size, memaddr);
+  uint16_t size = buffer_size;
+  uint8_t *p = buff;
+  i2c::ErrorCode ret;
+  while (size >= 1) {
+    // Limit the amount to read to a page size
+    uint16_t amt_to_read = size;
+    if (amt_to_read > this->i2c_buffer_size_)  // I2C buffer size limit
+      amt_to_read = this->i2c_buffer_size_;
+
+    // Check if we are dealing with large (>512kbit) EEPROMs
+    uint8_t i2c_address = this->address_;
+    if (this->get_memory_size_() > 0xFFFF) {
+      // Figure out if we are going to cross the barrier with this read
+      if (memaddr < 0xFFFF) {
+        if (0xFFFF - memaddr < amt_to_read)  // 0xFFFF - 0xFFFA < I2C_buffer_size
+          amt_to_read = 0xFFFF - memaddr;    // Limit the read amt to go right up to edge of barrier
+      }
+
+      // Figure out if we are accessing the lower half or the upper half
+      if (memaddr > 0xFFFF)
+        i2c_address |= 0b100;  // Set the block bit to 1
+    }
+    if (this->get_address_size_bytes_() == 2) {
+      uint8_t maddr[] = {(uint8_t) (memaddr >> 8), (uint8_t) (memaddr & 0xFF)};
+      ret = this->bus_->write(i2c_address, maddr, 2, false);
+    } else {
+      uint8_t maddr[] = {(uint8_t) (memaddr & 0xFF)};
+      ret = this->bus_->write(i2c_address, maddr, 1, false);
+    }
+    if (ret != i2c::ERROR_OK) {
+      ESP_LOGE(TAG, "Read raise this error %d on setting address", ret);
+    }
+    ESP_LOGVV(TAG, "Read - Done Set address, Ammount to read %d", amt_to_read);
+    ret = this->bus_->read(i2c_address, p, amt_to_read);
+    if (ret != i2c::ERROR_OK) {
+      ESP_LOGE(TAG, "Read raised this error %d on reading data", ret);
+    }
+    ESP_LOGVV(TAG, "Done Read");
+    memaddr += amt_to_read;
+    p += amt_to_read;
+    size -= amt_to_read;
+  }
+}
+/// @brief Read a std::string from the device
+/// @note It write the string with an extra byte containing the size at the memaddr
+/// @note It is limited to reading a max length of 254 bytes
+/// @param memaddr is the starting location to read
+/// @param str_to_read will hold the bytes read from the device on return of the fuction
+
+uint32_t ExtEepromComponent::read_string_from_eeprom(uint32_t memaddr, std::string &str_to_read) {
+  uint8_t new_str_len = read8(memaddr);
+  uint8_t data[new_str_len + 1];
+  read(memaddr + 1, (uint8_t *) data, new_str_len);
```

Comment:
```suggestion
  this->read(memaddr + 1, (uint8_t *) data, new_str_len);
```
I see a few more of these below -- please be sure to include the `this->` prefix.
Sorry for not catching this earlier!

Reply:
Have updated the code to resolve this and the issues below. Thank You